### ISSUE TITLE
Updating the ts file of the project translation context

### ISSUE THREAD

**Issue by ghtmtt** [qgis/QGIS#38767]
**Feature description.**

With the current translations framework if the whole project is already translated and I add other layers or change the field names (you get what I mean), the only way to not loose the existing translations is to use `lconvert` in the command line.

It would be nice if in the Project Properties, if the project has already be translated, to have a button like "Refresh Translations".

Hoping this make sense @m-kuhn :)

**Comment by m-kuhn** [comment]
What we do is exporting a "master" language version (no translations inside, only sources). This can be overwritten from qgis. The translated versions (managed here in transifex but could be qtlinguist) are not overwritten by qgis.

It would be good if you could add more details about your workflow and which file is overwritten at what point.

**Comment by ghtmtt** [comment]
The workflow is:

* QGIS locale = en
* generating the ts file from QGIS in en
* translating the strings in other languages and save both ts and qm file as `project_it.ts` `project_it.qm`
* adding other layers in the project

if now I generate another ts file from QGIS and want to translate it, all good, but I loose all the previous translations, so

* `lconvert new_ts_file_generated.ts project_it.ts -o new_updated_project_file.ts`
* `new_updated_project_file.ts` contains all the strings BUT I only have to translate the new strings, all the old ones are already there

Is anything wrong in this workflow?

**Comment by m-kuhn** [comment]
> if now I generate another ts file from QGIS and want to translate it, all good, but I loose all the previous translations, so

can you elaborate more about this step, how do you want to translate it?
I would assume that your translation tools take care of the merge.
If that's not the case... Should QField check for all translation versions (_it, _de, _fr...) next to the source language (_en) and create merged versions of all of them?

**Comment by ghtmtt** [comment]
mmm what I mean is that when modifying something in the project (adding a layer, etc) and the ts file has already been generated and translated in different languages, the `lconvert` tool merges automatically the translated ts file with the new one and you only have to translated the new strings.

I mean, that's OK, but it might be a good solution to let QGIS handle this.

* create a project and generate ts file
* translate the ts file in many languages and release them as qm
* opening the project QGIS with a slave language and OK
* opening th project QGIS with the master language and change something

now:

* is the project already configured with translations?
* YES -> check for existing translated strings and only tag as untranslated the ones with the changes

Does it make sense?

**Comment by m-kuhn** [comment]
> translate the ts file in many languages and release them as qm

That's where I would expect this merge magic to happen. It depends on this tool, what it does. If it doesn't happen automatically your proposal makes sense.

**Comment by ghtmtt** [comment]
exactly. I don't see that this is happening automatically. Should I make other tests?

**Comment by m-kuhn** [comment]
I cannot think of anything as my primary use case was online translation services. We can also leave this as feature request.

**Comment by RobiFag** [comment]
Hi all!
I shared a python script about the update of ts file, it is a bit rough but I hope it could help. You can find it here
https://github.com/qgis/QGIS-Enhancement-Proposals/issues/90#issuecomment-575675111
